backend/trade/views.py

```diff
@@ -21,39 +21,64 @@ class TradeListingViewSet(viewsets.ModelViewSet):
 
     def get_queryset(self):
         user = self.request.user
-        return Trade.objects.filter(models.Q(initiator=user) | models.Q(recipient=user)
-                                    ).select_related('initiator', 'recipient')
+        return Trade.objects.filter(
+            models.Q(initiator=user) | models.Q(recipient=user)
+        ).select_related('initiator', 'recipient')
 
     def perform_create(self, serializer):
         serializer.save(initiator=self.request.user, trade_status=Trade.NEGOTIATE)
 
-    @action(detail=True, methods=['post'])
+    @action(detail=True, methods=['put'])
     def negotiate(self, request, pk=None):
+        """
+        Handles negotiation:
+        - Automatically updates the negotiator's decision if listings or cash have changed.
+        - Resets the other participant's decision to "pending."
+        """
         trade = get_object_or_404(Trade, pk=pk)
 
         if trade.trade_status in [Trade.ACCEPTED, Trade.REJECTED]:
-            return Response({'detail': 'Cannot negotiate after trade is accepted or rejected.'},
+            return Response({'detail': 'Cannot negotiate after trade is finalized.'},
                             status=status.HTTP_400_BAD_REQUEST)
 
-        if request.user != trade.initiator and request.user != trade.recipient:
+        if request.user not in [trade.initiator, trade.recipient]:
             return Response({'detail': 'You do not have permission to negotiate this trade.'},
                             status=status.HTTP_403_FORBIDDEN)
 
-        serializer = self.get_serializer(trade, data=request.data, partial=True)
+        # Check if user modifies cash or listings
+        data = request.data
+        changed = False
+
+        if 'initiator_listing' in data or 'recipient_listing' in data or 'initiator_cash' in data or 'recipient_cash' in data:
+            changed = True
+
+        if changed:
```

Comment:
Small one, but I would delete lines 50-54 and just put line 52 for this check

Reply:
Done